### PR TITLE
Fix directory loader apparition in toolbar, and elements shown before metada is ready

### PR DESCRIPTION
**Pull request by sBouzols**
Signed-off-by: sBouzols <sylvain.bouzols@gmail.com>

### CODE REVIEW

**Review by JacquesBorsenberger — APPROVED** [review]

File: src/components/directory-content.js [hunk 1]
```diff
@@ -568,92 +569,113 @@ const DirectoryContent = () => {
...
+            </>
+        );
+    };
+
+    const renderContent = () => {
+        if (isMissingDataAfterDirChange) {
+            return renderLoadingContent();
+        } else {
+            if (currentChildren && selectedDirectory) {
+                console.log('SBO currentChildren', currentChildren);
```
Comment (laurgarn) [inline src/components/directory-content.js]: Remove console.log

File: src/components/directory-content.js [hunk 2]
```diff
@@ -568,92 +569,113 @@ const DirectoryContent = () => {
...
+                if (currentChildren.length === 0) {
+                    return renderEmptyDirContent();
+                } else {
+                    return renderTableContent();
+                }
+            }
+            return;
```
Comment (laurgarn) [inline src/components/directory-content.js]: Why this useless return at end of block ?
I'd go for keeping the `if`s aligned, with commenting/stating why `!currentChildren` and `!selectedBranches` alternatives should simply not show.

**Review by laurgarn — APPROVED** [review]

File: src/components/directory-content.js (hunk 2 above)
Comment (sBouzols) [reply src/components/directory-content.js]: fixed

File: src/components/directory-content.js (hunk 1 above)
Comment (sBouzols) [reply src/components/directory-content.js]: removed

**Pull request merged**